fs/fcb/src/fcb.c

```diff
@@ -226,8 +247,14 @@ fcb_offset_last_n(struct fcb *fcb, uint8_t entries,
         struct fcb_entry *last_n_entry)
 {
     struct fcb_entry loc;
+    int rc = 0;
     int i;
 
+    rc = os_mutex_pend(&fcb->f_mtx, OS_WAIT_FOREVER);
+    if (rc && rc != OS_NOT_STARTED) {
+        return FCB_ERR_ARGS;
+    }
+
```

Comment:
there is fcb_getnext_nolock() which could be used in loop below to avoid nested locking